src/Dialect/ONNX/ONNXShapeHelper.cpp

```diff
@@ -168,3 +168,7 @@ LogicalResult ONNXSliceOpShapeHelper::Compute(
   }
   return success();
 }
+
+namespace ONNXShapeFoldHelper {
+std::map<Operation *, SmallVector<int64_t, 4>> shapeMap;
```

Comment:
Worth adding a comment here explaining what this is for

Reply:
Also, what happens in the case where ShapeInference is ran multiple times throughout compilation?
E.g.:
ShapeInference
ConstantPropagation
ShapeInference

Won't this map potentially have invalid entries on subsequent runs?

Reply:
In ShapeInferencePass.cpp at the end of each pass run this structure is cleared. Also I am adding comments to this structure.

Reply:
Looks like it might not be possible with the erase from the map at the end, seems fine.

Reply:
@syin-groq Could we add assertion of empty map at the beginning and end of ShapeInference?

Reply:
@dhan-groq yes will do